lib/core/src/Cardano/Wallet/Primitive/CoinSelection/Balance.hs

```diff
@@ -233,12 +235,21 @@ emptySkeleton = SelectionSkeleton
 
 -- | Specifies a limit to adhere to when performing a selection.
 --
-data SelectionLimit
+type SelectionLimit = SelectionLimitOf Int
+
+data SelectionLimitOf a
     = NoLimit
       -- ^ Indicates that there is no limit.
-    | MaximumInputLimit Int
+    | MaximumInputLimit a
       -- ^ Indicates a maximum limit on the number of inputs to select.
-    deriving (Eq, Show)
+    deriving (Eq, Functor, Show)
+
+instance Ord a => Ord (SelectionLimitOf a) where
```

Comment:
Just to confirm, the Ord instance of SelectionLimitOf treats NoLimit is treated like "infinity"? If so, this instance makes sense.

Reply:
Yes, that's correct!

For completeness (and to satisfy the `Ord` laws), we must also state that `NoLimit == NoLimit`.